pkg/appmanager/appManager_test.go

```diff
@@ -2464,6 +2464,15 @@ var _ = Describe("AppManager Tests", func() {
 				Expect(r).To(BeTrue(), "Ingress resource should be processed.")
 				Expect(resources.Count()).To(Equal(0))
 
+				// Shouldn't process Ingress with non-F5 class
+				ingressNotf5 := test.NewIngress("ingress-bad", "1", namespace, ingressConfig,
+					map[string]string{
+						"kubernetes.io/ingress.class": "notf5",
+					})
+				r = mockMgr.addIngress(ingressNotf5)
+				Expect(r).To(BeFalse(), "Ingress resource should not be processed.")
+				Expect(resources.Count()).To(Equal(0))
+
```

Comment:
What are your thoughts on also testing the "modify existing ingress.class annotation" case?

I added this code here and it worked (and I'd sleep better at night if we had some test case like this):

```
ingressNotf5.Annotations["kubernetes.io/ingress.class"] = "f5"
r = mockMgr.updateIngress(ingressNotf5)
Expect(r).To(BeTrue(), "Ingress resource should be processed when flipping from notf5 to f5")
Expect(resources.Count()).To(Equal(1))
ingressNotf5.Annotations["kubernetes.io/ingress.class"] = "notf5again"
r = mockMgr.updateIngress(ingressNotf5)
Expect(r).To(BeFalse(), "Ingress resource should be destroyed when flipping from f5 to notf5again")
Expect(resources.Count()).To(Equal(0))
```